src/Gaufrette/Adapter/Ftp.php

```diff
@@ -366,7 +366,12 @@ private function isDir($directory)
             return true;
         }
 
-        if (!@ftp_chdir($this->getConnection(), $directory)) {
```

Comment:
Thank you for your contribution :)

From what I read on the php doc [1], the usage of `is_dir` with an URI having the `ftp`/`sftp` protocol is only allowed in passive mode. By default, the passive mode is disabled (c.f. the `__contructor` of this adapter).

From your PR description, it looks like you have an exception. What about to still use the `ftp_chdir` function, but catch the exception to fallback to `is_dir` ?

Also, can you create a dedicated method to compute the connection string ?

[1] https://php.net/manual/en/wrappers.ftp.php

Reply:
Thanks for these tips!
I created a private method for the connection URL.
I also kept the `chdir` method as the main approach and fallback on `is_dir` with passive mode set to `true` if we catch and exception with `chdir`.
Let me know if this looks better to you.